File: doc/src/scripts/InfernalRoboticsExample.cpp
#include <iostream>
#include <vector>
#include <krpc.hpp>
#include <krpc/services/space_center.hpp>
#include <krpc/services/infernal_robotics.hpp>

using SpaceCenter = krpc::services::SpaceCenter;
using InfernalRobotics = krpc::services::InfernalRobotics;

int main() {
  auto conn = krpc::connect("InfernalRobotics Example");
  SpaceCenter space_center(&conn);
  InfernalRobotics infernal_robotics(&conn);

  InfernalRobotics::ServoGroup group = infernal_robotics.servo_group_with_name(space_center.active_vessel(), "MyGroup");
  if (group == InfernalRobotics::ServoGroup())
    std::cout << "Group not found" << std::endl;

  std::vector<InfernalRobotics::Servo> servos = group.servos();
  for (auto servo : servos)
    std::cout << servo.name() << " " << servo.position() << std::endl;

  group.move_right();
  sleep(1);
  group.stop();
}

File: doc/src/scripts/InfernalRoboticsExample.py
import time
import krpc

conn = krpc.connect(name='InfernalRobotics Example')
vessel = conn.space_center.active_vessel

group = conn.infernal_robotics.servo_group_with_name(vessel, 'MyGroup')
if group is None:
    print('Group not found')
    exit(1)

for servo in group.servos:
    print servo.name, servo.position

group.move_right()
time.sleep(1)
group.stop()
